internal/provider/resource_address_lot_test.go

```diff
@@ -0,0 +1,114 @@
+package provider
+
+import (
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+)
+
+func TestAccAddressLot_full(t *testing.T) {
+	resourceName := "oxide_address_lot.test"
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:                 func() { testAccPreCheck(t) },
+		ProtoV6ProviderFactories: testAccProtoV6ProviderFactories(),
+		Steps: []resource.TestStep{
+			{
+				Config: testResourceAddressLotConfig,
+				Check:  checkResourceAddressLot(resourceName),
+			},
+			{
+				Config: testResourceAddressLotUpdateConfig,
+				Check:  checkResourceAddressLotUpdate(resourceName),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+var testResourceAddressLotConfig = `
+resource "oxide_address_lot" "test" {
+	description       = "a test address lot"
+	name              = "terraform-acc-my-address-lot"
+	kind              = "infra"
+	blocks = [
+		{
+			first_address = "172.0.1.1"
+			last_address  = "172.0.1.10"
+		},
+	]
+	timeouts = {
+		read   = "1m"
+		create = "3m"
+		delete = "2m"
+		update = "4m"
+	}
+}
+`
+
+var testResourceAddressLotUpdateConfig = `
+resource "oxide_address_lot" "test" {
+	description       = "a test address lot"
+	name              = "terraform-acc-my-address-lot"
+	kind              = "infra"
+	blocks = [
+		{
+			first_address = "172.0.1.1"
+			last_address  = "172.0.1.10"
+		},
+		{
+			first_address = "172.0.10.1"
+			last_address  = "172.0.10.10"
+		},
+	]
```

Comment:
Hm, you do already have multiple blocks in the test. I wonder if truly the nested computed attribute in sets is only an issue when it's nested deeper than the top level. The testing I asked for in another comment would show more.